app/mersea-front/src/components/Home.vue

```diff
@@ -49,13 +52,56 @@ export default {
         })
       )
       this.map.addControl(new mapboxgl.NavigationControl(), 'bottom-right')
+      this.map.on('load', async () => {
+        try {
+          let reports = await fetch(`${apiUrl}/reports`, {
+            method: 'get',
+            headers: new Headers({
+              Accept: 'application/geo+json',
+              'Content-Type': 'application/geo+json'
+            })
+          })
+          var geojson = await reports.json()
+        } catch (e) {
+          throw e
+        }
+
+        for (let marker of geojson.features) {
+          new mapboxgl.Marker({ color: marker.properties.color })
+            .setLngLat(marker.geometry.coordinates)
+            .setPopup(
+              new mapboxgl.Popup({ offset: 25 }).setHTML(`
+  <article class="media">
```

Comment:
Is it possible to make a Vue Component of this?

Reply:
I'll look into it

Reply:
I don't think we can make it work with a component since the popup needs a string. We would need to render the component somehow, parse the calculated html and stringify it.

Reply:
It's not compiled by Vue when mapbox add it to the DOM
https://github.com/phegman/vue-mapbox-gl#popups

Reply:
How about projects like this one https://soal.github.io/vue-mapbox?

Reply:
How about another project like this one https://github.com/soal/vue-mapbox? It looks like a more Vue-ish framework.

Reply:
I tried one at first https://github.com/phegman/vue-mapbox-gl but I'm not a fan adding an abstraction. I think it will add more blockers when coding advanced features